src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -229,11 +236,22 @@ private StudentParticipation createNewParticipationWithInitializationDate(Exerci
         participation.setInitializationState(InitializationState.UNINITIALIZED);
         participation.setExercise(exercise);
         participation.setParticipant(participant);
+
         // StartedDate is used to link a Participation to a test exam exercise
         if (initializationDate != null) {
             participation.setInitializationDate(initializationDate);
         }
-        return studentParticipationRepository.saveAndFlush(participation);
+        participation = studentParticipationRepository.saveAndFlush(participation);
+
+        ParticipationVCSAccessToken participationVCSAccessToken = new ParticipationVCSAccessToken();
+        if (exercise instanceof ProgrammingExercise && participant instanceof User && profileService.isLocalVcsActive()) {
```

Comment:
You could use something like `if (exercise instanceof ProgrammingExercise && participant instanceof User user && profileService.isLocalVcsActive()) {` instead of casting